### PR TITLE
Handled Max User Exceed

### PR DESCRIPTION
**Pull request by shakti97**
It will resolve and close issue #8

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -53,8 +53,8 @@ io.sockets.on('connection', socket => {
     socket.on('closeAuction', (namespace, owner_id) => {
         handleConnections.closeAuction(socket, io, namespace, owner_id);
     });
-    socket.on('joinRoom', (namespace, user_id) => {
-        handleConnections.joinAuction(socket, namespace, user_id);
+    socket.on('joinRoom', (namespace, user_id, max_user) => {
```
Comment (hariprasanths) [inline app.js]: Don't trust the clients to give this info
Associating this with admins is fine

File: assets/app/jsx/Auction.jsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React, {Component} from 'react';
 import dataFetch from './DataFetch';
 import io from 'socket.io-client';
+import Swal from 'sweetalert2';
```
Comment (hariprasanths) [inline assets/app/jsx/Auction.jsx:4]: Bootstrap has an inbuilt model, try using that
We would have one less package need

File: assets/app/jsx/Auction.jsx [hunk 3]
```diff
@@ -38,7 +40,37 @@ class Auction extends Component {
     componentDidMount() {
         socket.on('connect', () => {
             //once connected to server, join particular room
-            socket.emit('joinRoom', this.state.namespace, this.state.user_id);
+            dataFetch('/auctionConfig', {
```
Comment (hariprasanths) [inline assets/app/jsx/Auction.jsx]: Write a separate route in the same file instead of using q_type

File: assets/app/jsx/Auction.jsx [hunk 4]
```diff
@@ -38,7 +40,37 @@ class Auction extends Component {
...
+                        {
+                            max_users: response.message['max_users']
+                        },
+                        () => {
+                            socket.emit('joinRoom', this.state.namespace, this.state.user_id, this.state.max_users);
+                        }
+                    );
+                })
+                .catch(err => {
+                    console.log('Some Error Occured');
```
Comment (hariprasanths) [inline assets/app/jsx/Auction.jsx]: Use toastify to notify the user of the error

File: src/socketRoutes/connectionManager.js [hunk 5]
```diff
@@ -62,16 +62,20 @@ function closeAuction(socket, io, namespace, owner_id) {
...
-    socket.namespace = namespace;
+function joinAuction(socket, namespace, user_id, max_user) {
+    console.log('max_user ', max_user);
 
     //check is auction open
     if (clientSockets[namespace] === undefined) {
         //no room found (auction is closed or does not exist)
         socket.emit('auctionClosed', 'Auction is either closed or not open yet!');
+    } else if (Object.keys(clientSockets[namespace]).length >= max_user) {
+        console.log('max_limit_exceeded');
```
Comment (hariprasanths) [inline src/socketRoutes/connectionManager.js]: Remove all debug logs

**Review by hariprasanths — CHANGES_REQUESTED** [review]

File: app.js [hunk 6]
```diff
@@ -47,14 +47,14 @@ const handleBiding = require('./src/socketRoutes/bidManager');
...
+    socket.on('openAuction', (namespace, owner_id, max_user) => {
+        handleConnections.ownerSocket(socket, namespace, owner_id, max_user);
     });
     socket.on('closeAuction', (namespace, owner_id) => {
         handleConnections.closeAuction(socket, io, namespace, owner_id);
     });
-    socket.on('joinRoom', (namespace, user_id) => {
-        handleConnections.joinAuction(socket, namespace, user_id);
+    socket.on('joinRoom', (namespace, user_id, max_user) => {
+        handleConnections.joinAuction(socket, namespace, user_id, max_user);
```
Comment (adarshPatel509) [inline app.js]: Remove max_users.  It's already send by adminSocket on openAuction.

File: assets/app/jsx/Auction.jsx [hunk 7]
```diff
@@ -38,7 +40,20 @@ class Auction extends Component {
     componentDidMount() {
         socket.on('connect', () => {
             //once connected to server, join particular room
-            socket.emit('joinRoom', this.state.namespace, this.state.user_id);
+            socket.emit('joinRoom', this.state.namespace, this.state.user_id, this.state.max_users);
```
Comment (adarshPatel509) [inline assets/app/jsx/Auction.jsx]: Remove max_users.

**Pull request merged**